tasks/encryption-at-rest.yml

```diff
@@ -56,49 +7,79 @@
     mode: 0755
     state: directory
 
-# This tasks helps is identifying if mariadb_encryption_keys has changed
-- name: "Generate keyfile.sha1"
-  ansible.builtin.copy:
-    content: "{{ mariadb_encryption_keys_string | hash('sha1') }}"
-    dest: "{{ mariadb_file_key_management_dir + '/' + mariadb_file_key_management_filename }}.sha1"
-    owner: mysql
-    group: mysql
-    mode: 0600
-    backup: true
-  when: mariadb_file_key_management_override or not _mariadb_fileenc_stat.stat.exists
-  register: _mariadb_keyssha1
+- name: Fail when keygenmode is manual and mariadb_encryption_keys or mariadb_encryption_password is not defined
+  ansible.builtin.fail:
+  when:
+    - not mariadb_encryption_keygen_auto
+    - (mariadb_encryption_keys is not defined) or (mariadb_encryption_password is not defined) or
+      (mariadb_encryption_keys | length <= 0) or (mariadb_encryption_password | length <=0)
+
+- name: "Stat {{ _mariadb_file_key_management_filekey_path }}"
+  ansible.builtin.stat:
+    path: "{{ mariadb_file_key_management_dir }}/{{ mariadb_file_key_management_filename }}"
+  register: _mariadb_filekeypass_stat
 
-# password_file
-- name: "Store the encryption password in {{ _mariadb_file_key_management_filekey_path }}"
-  ansible.builtin.copy:
-    content: "{{ mariadb_encryption_password }}"
-    dest: "{{ _mariadb_file_key_management_filekey_path }}"
+- name: "Create encrypted file when user set its own keys"
```

Comment:
This task copies the user provided and encryption keys and password on the remote host. The **encrypted file** is created with the last task in this file using the openssl binary.


Therefore i suggest renaming this task to something like **Store user encryption keys and password locally** to reflect accurately what the task does.